File: src/SimpleStocker.Api/Models/Entities/SaleItem.cs
namespace SimpleStocker.Api.Models.Entities
{
    public class SaleItem : BaseEntity
    {
        public long SaleId { get; set; }
        public long ProductId { get; set; }
        public double Quantity { get; set; }
        public decimal UnityPrice { get; set; }
        public decimal SubTotal { get => (decimal)Quantity * UnityPrice; }
    }
}

File: src/SimpleStocker.Web/src/shared/models/SaleItem.ts
export class SaleItem {
  id: number
  createdDate: string
  updatedDate: string
  saleId: number
  productId: number
  quantity: number
  unityPrice: number
  subTotal: number

  constructor(init?: Partial<SaleItem>) {
    this.id = init?.id ?? 0
    this.createdDate = init?.createdDate ?? ''
    this.updatedDate = init?.updatedDate ?? ''
    this.saleId = init?.saleId ?? 0
    this.productId = init?.productId ?? 0
    this.quantity = init?.quantity ?? 0
    this.unityPrice = init?.unityPrice ?? 0
    this.subTotal = init?.subTotal ?? 0
  }
}
